lib/OpenQA/CacheService/Client.pm

```diff
@@ -31,7 +32,27 @@ has attempts   => $ENV{OPENQA_CACHE_ATTEMPTS}           // 60;
 has sleep_time => $ENV{OPENQA_CACHE_ATTEMPT_SLEEP_TIME} // 5;
 has host       => sub { 'http://127.0.0.1:' . service_port('cache_service') };
 has cache_dir  => sub { $ENV{OPENQA_CACHE_DIR} || OpenQA::Worker::Settings->new->global_settings->{CACHEDIRECTORY} };
-has ua         => sub { Mojo::UserAgent->new(inactivity_timeout => 300) };
+has ua         => sub {
+    my $ua = Mojo::UserAgent->new(inactivity_timeout => 300);
+
+    # set PeerAddrInfo directly (in consistency with host property) to workaround getaddrinfo() being stuck in error
+    # state "Address family for hostname not supported" for local connections (see poo#78390#note-38).
+    # note: The socket_options function has only been added in Mojolicious 8.72. For older versions we still rely
+    #       on the monkey_patch within the BEGIN block of Worker.pm. It could be removed when we stop supporting older
+    #       Mojolicious versions.
+    my %cache_service_address = (
+        family   => AF_INET,
+        protocol => IPPROTO_TCP,
+        socktype => SOCK_STREAM,
+        addr     => pack_sockaddr_in(service_port('cache_service'), inet_aton('127.0.0.1')));
+    $ua->socket_options->{PeerAddrInfo} = [\%cache_service_address] if $ua->can('socket_options');
+    return $ua;
+};
+
+sub set_port ($self, $port) {
```

Comment:
Oh, you added signatures after i did my review. This might require a spec file update for the worker to also depend on `perl >= 5.20.0`.

Reply:
#3707